### PR TITLE
Improve metrics docs

### LINKED ISSUE #424: rename *_failed_request metrics
The redis_cluster_failed_request metric only records error messages received from the destination server.
Requests failing due to internal shotover errors are not recorded.

We should either:
* Rename it to redis_cluster_server_error_messages_received
* Change it to record every possible reason a request could fail

This also applies to all similar metrics in other transforms.

### PR DESCRIPTION
**Pull request by conorbros**
- Closes #424
- Clean up the code for the observability interface a bit (just renaming and moving stuff to be more appropriate)

### CODE REVIEW

File: docs/src/user-guide/configuration.md [hunk 1]
```diff
@@ -12,18 +12,24 @@ The configuration file is used to change general behavior of Shotover. Currently
...
 
-This is a single string that you can use to configure logging with Shotover. It supports [env_filter](https://docs.rs/env_logger/0.7.1/env_logger/) style configuration and filtering syntax. Log levels and filters can be dynamically changed while Shotover Proxy is still running.
+This is a single string that you can use to configure logging with Shotover. It supports [env_filter](https://docs.rs/env_logger/0.7.1/env_logger/) style configuration and filtering syntax. Log levels and filters can be dynamically changed while Shotover is still running.
 
-This can be done by a POST HTTP request to the `/filter` endpoint (configured located at the observability interface) with the env_filter string set as the POST data. For example:
+### observability_interface 
+
+Shotover has an observability interface for you to collect Prometheus data from. This value will define the address and port for Shotover's observability interface. It is configured as a string in the format of `127.0.0.1:8080` for IPV4 addresses or `[2001:db8::1]:8080` for IPV6 addresses.
+
+#### Metrics
```
Comment (rukai) [inline docs/src/user-guide/configuration.md:21]: Shouldnt these `####` level headings go in observability.md?

Comment (conorbros) [reply docs/src/user-guide/configuration.md]: Sure, I've moved them now.

**Review by rukai — APPROVED** [review]
new docs are useful and the code cleanup makes sense

**Review by Claude-at-Instaclustr — APPROVED** [review]
Looks good to me

**Pull request merged**